src/agent/border_agent.cpp

```diff
@@ -217,19 +217,19 @@ static const char *ThreadVersionToString(uint16_t aThreadVersion)
 
 void BorderAgent::PublishService(void)
 {
+    Mdns::Publisher::TxtList txtList;
```

Comment:
Maybe we can use brace initialization when constructing the `TxtList` rather than `emplace_back`?
```
Mdns::Publisher::TxtList txtList { {"nn", mNetworkName}, {"xp", mExtPanId, sizeof(mExtPanId)}, {"tv", versionString}};
```

Reply:
SGTM.